spring-integration-file/src/main/java/org/springframework/integration/file/remote/synchronizer/AbstractInboundFileSynchronizingMessageSource.java

```diff
@@ -144,6 +144,10 @@ public void afterPropertiesSet() throws Exception {
 					throw new FileNotFoundException(this.localDirectory.getName());
 				}
 			}
+			this.fileSource.setUseWatchService(true);
```

Comment:
So, let's reinstate our old `RecursiveLeafOnlyDirectoryScanner` and provide an option to the remote `AbstractInboundFileSynchronizingMessageSource` to let to inject `DirectoryScanner` impl?!